cpp/src/hash/concurrent_unordered_multimap.cuh

```diff
@@ -539,6 +539,9 @@ public:
         const size_type partition_size  = m_hashtbl_size/num_parts;
 
         // Only insert into the specified partition
+	// Note that if m_hashtbl_size % num_parts != 0 then
+	// hash_tbl_idx/partition_size can exceed num_parts-1 for the last few
+	// elements, so we map them to the previous num_parts-1 partition
```

Comment:
I think it'd be better to use `get_partition` here instead of recomputing `hash_tbl_idx/partition_size` twice. Then you know that the assigned partition number is already a valid partition number.